update docs for surveys (#6218)
PostHog/posthog.com: contents/docs/surveys/manual.mdx

### Option 1 (recommended)

> 1. Make sure you've enabled and are up to date with the surveys app in your project's apps.

Author: ohhh cool even better

> **Note:** You'll need to handle whether a user has seen a survey already or not manually if you're using this option. The [survey app](https://github.com/PostHog/feature-surveys/blob/main/site.ts) uses `localStorage` to keep track of this, which you can implement similarly in your code.
> **Note:** If you're using a URL or selector targeting option, you'll need to poll with `posthog.getActiveMatchingSurveys()` to get the most updated survey results.

3. In order for survey responses to be recorded, you'll need to send in a "survey sent" capture call event with the survey's name, ID, and any properties you'd want.

```js-web
posthog.capture("survey sent", {
    $survey_id: {survey.id} // required
    $survey_response: {survey_response} // required
    $survey_name: {survey.name} // optional
})
```

Surveys that are shown should be captured with a "survey shown" event and those that are dismissed, with a "survey dismissed" event similar to the above.

Right now we only support one page open feedback and external link type surveys, so be sure to follow the syntax above to use our built in results functionality for API surveys. We'll be adding more types soon!

## Creating a survey

> To create a new survey, go to the surveys page in PostHog and click "New survey". You'll be prompted to enter a name for your survey, optional description, and what type of survey you'd like to create.

Reviewer: ```suggestion
To create a new survey, go to the surveys page in PostHog and click "New survey". You'll be prompted to enter a name for your survey, optional description, and what type of survey you'd like to create.

```

> To create a new survey, go to the surveys page in PostHog and click "New survey". You'll be prompted to enter a name for your survey, optional description, and what type of survey you'd like to create.

Reviewer: ```suggestion
To create a new survey, go to the Surveys page in PostHog and click "New survey". You'll be prompted to enter a name for your survey, optional description, and what type of survey you'd like to create.
```

On the right side of the page, you'll see a preview of what your survey will look like, along with appearance customization options.

![Create a survey](/docs/img/surveys/surveys-new-survey.png)

### Survey targeting

When you create a survey, it'll save as a draft until you're ready to launch it. You can target your survey to specific users based on:

> 1. Page URL contains. You can target a survey to a specific page URL, or a page URL that contains a specific string. For example, you can target a survey to `https://posthog.com/docs` or `https://posthog.com/docs/*` with the string "/docs".

Reviewer: ```suggestion
1. **Page URL contains.** You can target a survey to a specific page URL, or a page URL that contains a specific string. For example, you can target a survey to `https://posthog.com/docs` or `https://posthog.com/docs/*` with the string "/docs".
```

> 2. Selector. You can target a survey based on whether a specific element appears on your page with either class name or ID. For example, you can target a survey with `#my-button` or `.my-button`.

Reviewer: ```suggestion
2. **Selector.** You can target a survey based on whether a specific element appears on your page with either class name or ID. For example, you can target a survey with `#my-button` or `.my-button`.
```

> 3. Linked feature flag. You can target a survey to users who have a specific feature flag enabled. For example, you're rolling out a new feature flag "new-landing-page" and want to gather feedback only from users who have that flag enabled.

Reviewer: ```suggestion
3. **Linked feature flag.** You can target a survey to users who have a specific feature flag enabled. For example, you're rolling out a new feature flag "new-landing-page" and want to gather feedback only from users who have that flag enabled.
```

> 4. User/group properties. You can target a survey to users who have a specific user or group property. For example, you can target a survey to users who have a property `is_paying` set to `true`.

Reviewer: ```suggestion
4. **User/group properties.** You can target a survey to users who have a specific user or group property. For example, you can target a survey to users who have a property `is_paying` set to `true`.
```

> All targeting properties behave in a "AND" manner, meaning that a survey will only be shown to a user if _all_ of the above conditions are met.

![Survey targeting](/docs/img/surveys/surveys-targeting-options.png)

## Survey events

> When survey responses are collected, they're simply stored as PostHog events. We provide an easy way for you to view the responses on the survey page, but you can also view them in any manner you'd like with the PostHog UI.

Reviewer: ```suggestion
Survey responses are stored as PostHog events. We provide an easy way for you to view the responses on the survey page, but you can also view them in any manner you'd like with the PostHog UI.
```

    $survey_response: {survey_response} // required
    $survey_name: {survey.name} // optional
})

> ```

Author: I'll add more "API" mode tips once I've set up the handling for that type on the UI too! It should be in its own page, along with "how to poll" and etc

> ```

Author: left this on the manual instead

## Creating a survey

To create a new survey, go to the surveys page in PostHog and click "New survey". You'll be prompted to enter a name for your survey, optional description, and what type of survey you'd like to create.

> On the right side of the page, you'll see a preview of what your survey will look like along with appearance customization options.

Reviewer: ```suggestion
On the right side of the page, you'll see a preview of what your survey will look like, along with appearance customization options.
```

![Create a survey](/docs/img/surveys/surveys-new-survey.png)

### Survey targeting

When you create a survey, it'll automatically save as a draft until you're ready to launch it. You can target your survey to specific users based on:

> 1. Page url contains. You can target a survey to a specific page url, or a page url that contains a specific string. For example, you can target a survey to `https://posthog.com/docs` or `https://posthog.com/docs/*` with the string "/docs".

Reviewer: ```suggestion
1. Page URL contains. You can target a survey to a specific page URL, or a page URL that contains a specific string. For example, you can target a survey to `https://posthog.com/docs` or `https://posthog.com/docs/*` with the string "/docs".
```

Acronym (Unified Resource Locator)

2. Selector. You can target a survey based on whether a specific element appears on your page with either class name or ID. For example, you can target a survey with `#my-button` or `.my-button`.

> 3. Linked feature flag. You can target a survey to users who have a specific feature flag enabled. For example, you're rolling out a new feature flag "new-landing-page" and you want to gather feedback from users who have that flag enabled.

Reviewer: ```suggestion
3. Linked feature flag. You can target a survey to users who have a specific feature flag enabled. For example, you can roll out a new feature flag "new-landing-page" and gather feedback from users who have that flag enabled.
```

4. User/group properties. You can target a survey to users who have a specific user or group property. For example, you can target a survey to users who have a property `is_paying` set to `true`.

> All of the above properties behave in a "AND" manner, meaning that a survey will only be shown to a user if all of the above conditions are met.

Reviewer: ```suggestion
> All targeting properties behave in a "AND" manner, meaning that a survey will only be shown to a user if _all_ of the above conditions are met.
```
Feel this is worth highlighting

![Survey targeting](/docs/img/surveys/surveys-targeting-options.png)

## Survey events

When survey responses are collected, they're simply stored as PostHog events. We provide an easy way for you to view the responses on the survey page, but you can also view them in any manner you'd like with the PostHog UI.

> We have "survey shown", "survey dismissed", and "survey sent" events. All three events will contain the `$survey_id` and `$survey_name` properties, and "survey sent" will also contain the `$survey_response` property.

Reviewer 2: We need to update Option 2 to tell that they'll have to send these events themselves as well.

Author: yesss! will do for the "headless api mode" docs page

Author: left this on the manual instead
